test/NuGet.Core.Tests/NuGet.Packaging.Test/ContentModelTests/ReadOnlyMemoryCharComparerOrdinalTests.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using FluentAssertions;
+using NuGet.ContentModel;
+using Xunit;
+
+namespace NuGet.Packaging.Test.ContentModelTests
+{
+    public class ReadOnlyMemoryCharComparerOrdinalTests
+    {
+        [Theory]
+        [InlineData("ABCDEFG", "ABCDEFG", true)]
+        [InlineData("ABCDEFG", "abcdefg", false)]
+        [InlineData("ref/four/foo.dll", "lib/four/bar.dll", true, 4, 4)]
+        [InlineData("ref/four/foo.dll", "lib/five/bar.dll", false, 4, 4)]
+        public void Equals_ReturnsExpectedValue(string x, string y, bool expected, int? start = default, int? length = default)
```

Comment:
I get what the two tests that use `start` and `length` are going for, but IMO it's unnecessary because it's really testing `AsMemory`'s implementation, not `ReadOnlyMemoryCharComparerOrdinal`'s implementation.

Reply:
Great point. Removed.
I also added more test cases in light of your comments on the hash code implementation.